server/src/main/java/org/elasticsearch/common/settings/LocallyMountedSecrets.java

```diff
@@ -78,7 +89,8 @@ public class LocallyMountedSecrets implements SecureSettings {
     public LocallyMountedSecrets(Environment environment) {
         var secretsDirPath = resolveSecretsDir(environment);
         var secretsFilePath = resolveSecretsFile(environment);
-        secretsParser.declareObject(ConstructingObjectParser.constructorArg(), (p, c) -> p.map(), SECRETS_FIELD);
+        secretsParser.declareObject(ConstructingObjectParser.optionalConstructorArg(), (p, c) -> p.map(), STRING_SECRETS_FIELD);
```

Comment:
Don't we need to do this in a bwc way so that control plane has a chance to convert to the new key name?

Reply:
I made `secrets` a deprecated name for this argument:

```
    public static final ParseField STRING_SECRETS_FIELD = new ParseField("string_secrets", "secrets");
```

I also set up the parser to ignore deprecations:

```
var parser = JSON.xContent()
                .createParser(XContentParserConfiguration.EMPTY.withDeprecationHandler(DeprecationHandler.IGNORE_DEPRECATIONS), bis)
```

There's a test that verifies that we can still load files using the old key name. So I think we're _almost_ covered here, except we might want to log a deprecation warning.

Reply:
> I made secrets a deprecated name

I glossed over that, thanks for pointing it out!

> except we might want to log a deprecation warning

I don't think a deprecation warning will do anything for us. We don't keep the deprecation log around in serverless.